aws/resource_aws_ecr_lifecycle_policy.go

```diff
@@ -0,0 +1,101 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsEcrLifecyclePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEcrLifecyclePolicyCreate,
+		Read:   resourceAwsEcrLifecyclePolicyRead,
+		Delete: resourceAwsEcrLifecyclePolicyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"repository": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"policy": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateJsonString,
+			},
+			"registry_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEcrLifecyclePolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecrconn
+
+	input := &ecr.PutLifecyclePolicyInput{
+		RepositoryName:      aws.String(d.Get("repository").(string)),
+		LifecyclePolicyText: aws.String(d.Get("policy").(string)),
+	}
+
+	resp, err := conn.PutLifecyclePolicy(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.RepositoryName)
+	d.Set("registry_id", *resp.RegistryId)
```

Comment:
We tend to intentionally avoid dereferencing pointers in `d.Set()` as we were burnt in the past when the API returned empty field and terraform ended up crashing because it tried to dereference `nil`. `Set()` can perform safe dereferencing, so it's ok to just pass the pointer, i.e. `d.Set("registry_id", resp.RegistryId)`

Reply:
I understand👍